libraries/utilities/log_expvars.py

```diff
@@ -37,61 +39,80 @@ def write_expvars(results_obj, endpoint):
+
+    try:
+
+        # Get gateload ips from ansible inventory
+        lgs_host_vars = hosts_for_tag(cluster_config, "load_generators")
+        lgs = [lg["ansible_host"] for lg in lgs_host_vars]
+        lgs_expvar_endpoints = [lg + ":9876/debug/vars" for lg in lgs]
+        log_info("Monitoring gateloads until they finish: {}".format("\n".join(lgs_expvar_endpoints)))
+
+        # Get sync_gateway ips from ansible inventory
+        sgs_host_vars = hosts_for_tag(cluster_config, "sync_gateways")
+        sgs = [sgv["ansible_host"] for sgv in sgs_host_vars]
+        sgs_expvar_endpoints = [sg + ":4985/_expvar" for sg in sgs]
+        log_info("Monitoring sync_gateways: {}".format("\n".join(sgs_expvar_endpoints)))
+
+        # Verify that sync gateway expvar endpoints are reachable
+        wait_for_endpoints_alive_or_raise(sgs_expvar_endpoints)
+
+        # Wait until the gateload expvar endpoints are up, or raise an exception and abort
+        wait_for_endpoints_alive_or_raise(lgs_expvar_endpoints)
+
+        start_time = time.time()
+        gateload_results = OrderedDict()
+        sync_gateway_results = OrderedDict()
+
+        gateload_is_running = True
+        sg_is_running = True
+        while gateload_is_running and sg_is_running:
+
+            # Caputure expvars for gateloads
+            for endpoint in lgs_expvar_endpoints:
+                try:
+                    log_info("Collecting gateload expavars {}".format(endpoint))
+                    write_expvars(gateload_results, endpoint)
+                    dump_results(folder_name, gateload_results, sync_gateway_results)
+                except RequestException as re:
+                    # connection to gateload expvars has been closed
+                    log_info(re)
+                    log_info("Error: {}.  Gateload {} no longer reachable. Writing expvars to {}".format(re, endpoint, folder_name))
+                    dump_results(folder_name, gateload_results, sync_gateway_results)
+                    gateload_is_running = False
+
+            # Capture expvars for sync_gateways
+            for endpoint in sgs_expvar_endpoints:
+                try:
+                    log_info("Collecting sg expavars {}".format(endpoint))
+                    write_expvars(sync_gateway_results, endpoint)
+                    dump_results(folder_name, gateload_results, sync_gateway_results)
+                except RequestException as re:
+                    # Should not happen unless sg crashes
+                    finished_successfully = False
+                    log_info(re)
+                    log_info("ERROR {}: sync_gateway not reachable. Dumping results to {}".format(re, folder_name))
+                    dump_results(folder_name, gateload_results, sync_gateway_results)
+                    sg_is_running = False
+
+            log_info("Elapsed: {} minutes".format((time.time() - start_time) / 60.0))
+            time.sleep(sleep_time)
+
+    except Exception as e:
```

Comment:
Changed to `RuntimeError` and pushed up to branch.

I'm afraid to make any more drastic changes because we don't have any unit tests for this code .. I'm relying on the real world tests I did